lib/ansible/modules/cloud/vmware/vmware_add_nic.py

```diff
@@ -0,0 +1,433 @@
+    nic_spec.device = nic_obj
+    nic_spec.device.key = nic_obj.key
+    nic_spec.device.macAddress = nic_obj.macAddress
+
+    if desired_nic['dvs']:
+        pg_obj = get_obj_by_name(conn, [vim.dvs.DistributedVirtualPortgroup], desired_nic['network'])
+        dvs_port_connection = vim.dvs.PortConnection()
+        dvs_port_connection.portgroupKey= pg_obj.key
+        dvs_port_connection.switchUuid= pg_obj.config.distributedVirtualSwitch.uuid
+        nic_spec.device.backing = vim.vm.device.VirtualEthernetCard.DistributedVirtualPortBackingInfo()
+        nic_spec.device.backing.port = dvs_port_connection
+    else:
+        nic_spec.device.backing = vim.vm.device.VirtualEthernetCard.NetworkBackingInfo()
+        nic_spec.device.backing.network = get_obj_by_name(conn, [vim.Network], desired_nic['network'])
+
+    nic_spec.device.connectable = vim.vm.device.VirtualDevice.ConnectInfo()
+    nic_spec.device.connectable.connected = True
+    nic_spec.device.connectable.startConnected = True
+    nic_spec.device.connectable.allowGuestControl = True
+    nic_spec.device.wakeOnLanEnabled = True
+    nic_spec.device.deviceInfo = vim.Description()
+    nic_spec.device.deviceInfo.summary = desired_nic['network']
+    nic_spec.device.backing.deviceName = desired_nic['network']
+
+    changes.append(nic_spec)
+    vm_spec.deviceChange = changes
+    task = vm.ReconfigVM_Task(spec=vm_spec)
+    success, result = wait_for_task(task)
+
+    if success:
+        return desired_nic
+    else:
+        module.fail_json(msg="Failed to update nic: %s" % result)
+
+
+def needs_update(module, desired_nic, all_nics):
+    nic_obj = None
+    all_nic_labels = []
+    matching_nics = []
+
+    for nic in all_nics:
+        all_nic_labels.append(nic['nic_obj'].deviceInfo.label)
+
+        if nic['nic_obj'].deviceInfo.label == desired_nic['label']:
+            matching_nics.append(nic['nic_obj'])
+
+    if len(matching_nics) == 1:
+        nic_obj = matching_nics[0]
+    elif len(matching_nics) > 1:
+        module.fail_json(msg="One or more NICs with the label: %s have been found. You'll have to update the proper NIC manually. Or ensure the NICs have unique labels and try again." % desired_nic['label'])
+    elif len(matching_nics) == 0:
+        module.fail_json(msg="No NIC with the name: %s was found. Found: %s " % (desired_nic['label'], all_nic_labels))
+
+
+    if nic_obj.backing.deviceName == desired_nic['network']:
+        if desired_nic['dvs']:
+            if 'Distrubted' in nic_obj.backing.__class__.__name__:
+                return False
+            else:
+                pass
+        else:
+            if 'Network' in nic_obj.backing.__class__.__name__:
+                return False
+            else:
+                pass
+
+    return True
+
+
+def main():
+
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        dict(
+                vm_path=dict(required=True, type='str'),
```

Comment:
Can you please include `uuid` as parameter for finding VM ? you can use `find_by_vm_id` API for finding VM using uuid just like other VMware modules do.